rules/aip0233/message_rules_test.go

```diff
@@ -0,0 +1,250 @@
+			problems := parentField.Lint(message.GetFile())
+			if diff := test.problems.SetDescriptor(problemDesc).Diff(problems); diff != "" {
+				t.Errorf(diff)
+			}
+		})
+	}
+}
+
+func TestRequestsField(t *testing.T) {
+	// Set up the testing permutations.
+	tests := []struct {
+		testName    string
+		src         string
+		problems    testutils.Problems
+		problemDesc func(m *desc.MessageDescriptor) desc.Descriptor
+	}{
+		{
+			testName: "Valid",
+			src: `
+message BatchCreateBooksRequest {
+	repeated CreateBookRequest requests = 1;
+}
+
+message CreateBookRequest {}
+`,
+			problems: testutils.Problems{},
+		},
+		{
+			testName: "Invalid-MissingRequestsField",
+			src: `
+message BatchCreateBooksRequest {
+	string parent = 1;
+}`,
+			problems: testutils.Problems{{Message: `Message "BatchCreateBooksRequest" has no "requests" field`}},
+		},
+		{
+			testName: "Invalid-RequestsFieldIsNotRepeated",
+			src: `
+message BatchCreateBooksRequest {
+	CreateBookRequest requests = 1;
+}
+message CreateBookRequest {}`,
+			problems: testutils.Problems{{Message: `The "requests" field should be repeated`}},
+			problemDesc: func(m *desc.MessageDescriptor) desc.Descriptor {
+				return m.FindFieldByName("requests")
+			},
+		},
+		{
+			testName: "Invalid-RequestsFieldWrongType",
+			src: `
+message BatchCreateBooksRequest {
+	repeated int32 requests = 1;
+}`,
+			problems: testutils.Problems{{Message: `The "requests" field on Batch Create Request should be a "CreateBookRequest" type`}},
+			problemDesc: func(m *desc.MessageDescriptor) desc.Descriptor {
+				return m.FindFieldByName("requests")
+			},
+		},
+		{
+			testName: "Invalid-RequestsNotRepeatedWrongType",
+			src: `
+message BatchCreateBooksRequest {
+	int32 requests = 1;
+}`,
+			problems: testutils.Problems{
+				{Message: `The "requests" field should be repeated`},
+				{Message: `The "requests" field on Batch Create Request should be a "CreateBookRequest" type`}},
+			problemDesc: func(m *desc.MessageDescriptor) desc.Descriptor {
+				return m.FindFieldByName("requests")
+			},
+		},
+	}
+
+	// Run each test individually.
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			file := testutils.ParseProto3String(t, test.src)
+
+			m := file.GetMessageTypes()[0]
+
+			// Determine the descriptor that a failing test will attach to.
+			var problemDesc desc.Descriptor = m
+			if test.problemDesc != nil {
+				problemDesc = test.problemDesc(m)
+			}
+
+			problems := requestsField.Lint(file)
+			if diff := test.problems.SetDescriptor(problemDesc).Diff(problems); diff != "" {
+				t.Errorf(diff)
+			}
+		})
+	}
+}
+
+func TestResourceField(t *testing.T) {
+	// Set up the testing permutations.
+	tests := []struct {
+		testName    string
+		src         string
+		problems    testutils.Problems
+		problemDesc func(m *desc.MessageDescriptor) desc.Descriptor
+	}{
+		{
+			testName: "Valid",
+			src: `
+message BatchCreateBooksResponse {
```

Comment:
It is possible to use templates here. See what I did in #304 on the AIP-231 ones.

Reply:
done.